server/src/test/java/io/crate/auth/HostBasedAuthenticationTest.java

```diff
@@ -545,12 +562,122 @@ public void cert_method_resolved_when_ssl_on_and_keystore_configured() throws Ex
             SystemDefaultDnsResolver.INSTANCE,
             () -> "dummy"
         );
+        SSLSession sslSession = mock(SSLSession.class);
+        when(sslSession.getPeerCertificates()).thenReturn(new Certificate[]{mock(Certificate.class)});
+
         AuthenticationMethod authMethod = hba.resolveAuthenticationType("crate",
-            new ConnectionProperties(InetAddresses.forString("1.2.3.4"), Protocol.TRANSPORT, mock(SSLSession.class)));
+            new ConnectionProperties(CRATE_USER_CREDENTIALS, InetAddresses.forString("1.2.3.4"), Protocol.TRANSPORT, sslSession));
         assertThat(authMethod).isExactlyInstanceOf(ClientCertAuth.class);
 
         AuthenticationMethod authMethod2 = hba.resolveAuthenticationType("crate",
-            new ConnectionProperties(InetAddresses.forString("1.2.3.4"), Protocol.TRANSPORT, mock(SSLSession.class)));
+            new ConnectionProperties(CRATE_USER_CREDENTIALS, InetAddresses.forString("1.2.3.4"), Protocol.TRANSPORT, sslSession));
         assertThat(authMethod2).isExactlyInstanceOf(ClientCertAuth.class);
     }
+
+    @Test
+    public void test_multiple_matches_method_uniquely_identified_by_properties() {
+        // Validating different declaration order as auth used to fail depending on order.
+        Settings pwdFirst = Settings.builder()
+            .put("auth.host_based.config.1.method", "password")
+            .put("auth.host_based.config.2.method", "jwt")
+            .put("auth.host_based.config.2.protocol", "http")
+            .build();
+        Settings jwtFirst = Settings.builder()
+            .put("auth.host_based.config.1.method", "jwt")
+            .put("auth.host_based.config.1.protocol", "http")
+            .put("auth.host_based.config.2.method", "password")
```

Comment:
```suggestion
```

Reply:
Sorry didn't get it - the whole point if this test is having 2 methods, if I remove password, I will end up with only jwt?